elfpy/utils/apeworx_integrations.py

```diff
@@ -167,92 +166,114 @@ def get_on_chain_trade_info(hyperdrive_contract: ContractInstance) -> OnChainTra
+        positive_balance = int(info.trades.loc[(trades_in_position) & (info.trades["to"] == address), "value"].sum())
+        negative_balance = int(info.trades.loc[(trades_in_position) & (info.trades["from"] == address), "value"].sum())
+        balance = positive_balance - negative_balance
+        logging.debug(f"balance {balance} = positive_balance {positive_balance} - negative_balance {negative_balance}")
         asset_prefix, maturity = hyperdrive_assets.decode_asset_id(position_id)
         asset_type = AssetIdPrefix(asset_prefix).name
         mint_time = maturity - elfpy.SECONDS_IN_YEAR
         log_and_show(f" => {asset_type}({asset_prefix}) maturity={maturity} mint_time={mint_time}")
+
+        on_chain_balance=0
         # verify our calculation against the onchain balance
-        on_chain_balance = hyperdrive_contract.balanceOf(position_id, address_)
-        if abs(balance - on_chain_balance) > elfpy.MAXIMUM_BALANCE_MISMATCH_IN_WEI:
-            raise ValueError(
-                f"events {balance=} and {on_chain_balance=} disagree by "
-                f"more than {elfpy.MAXIMUM_BALANCE_MISMATCH_IN_WEI} wei for {address_}"
-            )
-        log_and_show(f" => calculated balance = on_chain = {fmt(balance)}")
+        if add_to_existing_wallet is None: # sourcery skip: merge-nested-ifs
+            on_chain_balance = hyperdrive_contract.balanceOf(position_id, address)
+            # only do balance checks if not marignal update 
+            if abs(balance - on_chain_balance) > elfpy.MAXIMUM_BALANCE_MISMATCH_IN_WEI:
+                raise ValueError(
+                    f"events {balance=} and {on_chain_balance=} disagree by "
+                    f"more than {elfpy.MAXIMUM_BALANCE_MISMATCH_IN_WEI} wei for {address}"
+                )
+            log_and_show(f" => calculated balance = on_chain = {fmt(balance)}")
         # check if there's an outstanding balance
         if balance != 0 or on_chain_balance != 0:
             if asset_type == "SHORT":
                 # loop across all the positions owned by this wallet
                 sum_product_of_open_share_price_and_value, sum_value = 0, 0
                 for specific_trade in trades_in_position.index[trades_in_position]:
                     value = info.trades.loc[specific_trade, "value"]
-                    value *= -1 if info.trades.loc[specific_trade, "from"] == address_ else 1
+                    value *= -1 if info.trades.loc[specific_trade, "from"] == address else 1
                     sum_value += value
                     sum_product_of_open_share_price_and_value += (
                         value * info.share_price[info.trades.loc[specific_trade, "block_number"]]
                     )
-                open_share_price = sum_product_of_open_share_price_and_value / sum_value
+                open_share_price = int(sum_product_of_open_share_price_and_value / sum_value)
```

Comment:
just make sure you're aware that this scaled the value down when you cast to int.

Reply:
this could be an issue... IF the two inputs aren't already `int` or `float` which I think they are in this case (all our inputs are read in directly from the chain)